### ISSUE TITLE
Stopped working with v2.0.0

### ISSUE THREAD

**Issue by jeffschwartz** [AtomLinter/linter-jshint#214]
I'm running Atom v1.5.3 & after installing v2.0.0 jshint stopped working.

**Comment by steelbrain** [comment]
Can you please explain what you mean by stopped working? Also did you try restarting Atom? and can you try running jshint from the cli and show us what you see?

**Comment by jeffschwartz** [comment]
It isn't linting .js files - no errors and no warnings. It also doesn't lint when selecting lint from the command palate. Was working fine until the v2.0.0 update.

**Comment by steelbrain** [comment]
Does `apm install linter-jshint@1` really fix this for you? If it does, can you show us the output of `jshint .` from cli?

**Comment by jeffschwartz** [comment]
I installed using Atom's update page and not at the cl using apm. Are you requesting that I uninstall the plugin and then use the cli to install the plugin?

**Comment by steelbrain** [comment]
I'm asking you to confirm that there really is a bug with this update by temporarily rolling back to the previous stable version. and `jshint .` should execute jshint executable (which will give us an idea of what's wrong), please note that jshint executable and Atom package are two different things :)

**Comment by jeffschwartz** [comment]
Running from the command line:
~/Development/preamble-ts-standalone$ jshint dist/core/htmlReporter/htmlReporter.js
dist/core/htmlReporter/htmlReporter.js: line 15, col 9, Empty block.
dist/core/htmlReporter/htmlReporter.js: line 17, col 51, 'data' is defined but never used.

2 errors

**Comment by jeffschwartz** [comment]
Sure, I can do that and report back.

**Comment by jeffschwartz** [comment]
Getting this error when running apm install linter-jshint@1:
 ~/Development/preamble-ts-standalone$ apm install linter-jshint@1
Installing linter-jshint@1 to /Users/Jeff/.atom/packages ✗
Package version: 1 not found

**Comment by jeffschwartz** [comment]
I ran apm install linter-jshint@1.3.1 which worked.
Linting now works - errors are being flagged in the gutter as expected but tooltips along with other atom linter configuration options are not working so perhaps the issues is atom linter?

**Comment by jeffschwartz** [comment]
I update to v2.0.0 and errors are no longer being flagged using the same .js file as previously used with v1.3.1.

**Comment by jeffschwartz** [comment]
btw i'm running Atom on a Mac :)

**Comment by jeffschwartz** [comment]
One more piece of information: Atom Linter configuration options are all working with linter-tslint.

**Comment by eruizdechavez** [comment]
I am getting the same problem as @jeffschwartz, upgrading to v2 makes the issue reports to stop working completely. Downgrading to v1.3.1 restores functionality.

**Comment by SelvaRCH94** [comment]
I also face the same situation as  @eruizdechavez. I use win7.

**Comment by jeffschwartz** [comment]
Grimm, I failed to mention that the funky taste I was getting on the .2 coils disappeared after a few days of heavy use. I love the reduced amp drain/better battery life I get with .4 coils and hitting them with 45 watts and the airflow closed down to about half is the sweet spot for me. If I get the desire to vape at .2 ohms or lower I just go to my RDAs but so far, after 1 month of heavy use the Cleito has delivered the goods - great flavor and knock on wood, never a dry hit with the .4 coils. I'm sorry that you are experiencing issues with yours. Perhaps you can speak or communicate with someone from Aspire and let them know what you and others are experiencing. I am hopeful that your feedback would be well received. Now I am waiting on the RBA for this little puppy. Aspire has said it will be vertical so I am quite excited about that. While I really love the .4 coils so far, I hate spending all the $$$$ lol. Peace out!

**Comment by Arcanemagus** [comment]
Can anyone provide a file (+ configuration, if necessary) That this is reproducible on?

**Comment by eruizdechavez** [comment]
I guess I can help trace this, just let me know what you need.

**Comment by Arcanemagus** [comment]
@eruizdechavez A file to reproduce this with, or more detailed steps to reproduce since it's working for me on the spec files :stuck_out_tongue:

Does `apm test` show any errors when run from the package folder?

**Comment by eruizdechavez** [comment]
![linter-jshint](https://cloud.githubusercontent.com/assets/953140/13087475/21ae1b5c-d4b6-11e5-9bf8-72da0dd10249.gif)

**Comment by eruizdechavez** [comment]
@Arcanemagus `apm test` fails in for both `1.3.1` and `2`; I guess I need some package for atom development.

```sh
EChavez-MBP:linter-jshint erick.ruizdechavez$ apm test
[15786:0216/140906:ERROR:renderer_main.cc(200)] Running without renderer sandbox
[15782:0216/140907:INFO:CONSOLE(85)] "Error: Cannot find module '/Users/erick.ruizdechavez/.atom/packages/linter-jshint/spec'
  at Module._resolveFilename (module.js:336:15)
  at Function.Module._resolveFilename (/Applications/Atom.app/Contents/Resources/app.asar/src/module-cache.js:383:52)
  at Function.Module._load (module.js:286:25)
  at Module.require (module.js:365:17)
  at require (/Applications/Atom.app/Contents/Resources/app.asar/src/native-compile-cache.js:50:27)
  at requireSpecs (/Applications/Atom.app/Contents/Resources/app.asar/spec/jasmine-test-runner.js:101:7)
  at module.exports (/Applications/Atom.app/Contents/Resources/app.asar/spec/jasmine-test-runner.js:49:7)
  at module.exports (/Applications/Atom.app/Contents/Resources/app.asar/src/initialize-test-window.js:71:17)
  at setupWindow (file:///Applications/Atom.app/Contents/Resources/app.asar/static/index.js:86:5)
  at window.onload (file:///Applications/Atom.app/Contents/Resources/app.asar/static/index.js:41:9)
", source: /Applications/Atom.app/Contents/Resources/app.asar/src/initialize-test-window.js (85)
Tests failed
```

**Comment by Arcanemagus** [comment]
`apm test` fails since there aren't any implemented, sorry about mentioning that, was confusing this with the other JS linter that I don't use :stuck_out_tongue:.

Testing this out myself right now...

**Comment by Arcanemagus** [comment]
@eruizdechavez What version of Atom are you using? `linter-jshint@2.0.0` works perfectly fine for me on your test file :confused:

![image](https://cloud.githubusercontent.com/assets/427137/13087871/b4b0e190-d49e-11e5-9d3e-1de1d23140cc.png)

**Comment by Arcanemagus** [comment]
What is the output of `atom.config.get('linter-jshint.executablePath')` when you type that into Atom's Developer Tools?

**Comment by eruizdechavez** [comment]
Atom 1.5.3 

```sh
EChavez-MBP:linter-jshint erick.ruizdechavez$ apm -v
apm  1.6.0
npm  2.13.3
node 0.10.40
python 2.7.10
git 2.5.4
```

**Comment by eruizdechavez** [comment]
```sh
EChavez-MBP:linter-jshint erick.ruizdechavez$ apm ls | grep linter
├── linter@1.11.3
├── linter-jshint@2.0.0
├── minimap-linter@1.1.1
```

**Comment by eruizdechavez** [comment]
```sh
EChavez-MBP:~ erick.ruizdechavez$ npm ls -g | grep jshint
├─┬ jshint@2.8.0
```

**Comment by jeffschwartz** [comment]
atom.config.get('linter-jshint.executablePath')
"/Users/Jeff/.atom/packages/linter-jshint/node_modules/jshint/bin/jshint"

**Comment by Arcanemagus** [comment]
I can't reproduce on Atom 1.5.3 on Ubuntu either :confused:
![image](https://cloud.githubusercontent.com/assets/427137/13089036/b01c2842-d4a3-11e5-86c2-3d52099c9d8c.png)

(First screenshot was from Windows 10)

@jeffschwartz That looks correct...

**Comment by Arcanemagus** [comment]
Can you guys place a breakpoint [here](https://github.com/AtomLinter/linter-jshint/blob/v2.0.0/lib/main.js#L89) to see if the `lint()` function is even being called.

Place one [here](https://github.com/AtomLinter/linter-jshint/blob/v2.0.0/lib/main.js#L111), and see what `result` contains.

Place one [here](https://github.com/AtomLinter/linter-jshint/blob/v2.0.0/lib/main.js#L122) to see what `parsed` contains.

**Comment by Arcanemagus** [comment]
Ah ha! Do you guys have `Lint Inline Java Script` enabled? When that is enabled it shows no errors for that test file, there seems to be a pretty bad bug with that option from my investigation [here](https://github.com/AtomLinter/linter-jshint/issues/176#issuecomment-184367839).

**Comment by eruizdechavez** [comment]
Eureka! 

Yup, after disabling it, all starts working again.

**Comment by Arcanemagus** [comment]
As you can see, this is a bug in `linter-jshint`, but `jshint` itself:
![image](https://cloud.githubusercontent.com/assets/427137/13089333/fd0107e4-d4a4-11e5-9478-8c01ec236d16.png)

@jeffschwartz are you seeing the same thing?

**Comment by Arcanemagus** [comment]
@eruizdechavez If you are able to I would _highly_ recommend switching to a modern JS linter like [`eslint`](https://atom.io/packages/linter-eslint) or [`jscs`](https://atom.io/packages/linter-jscs).

**Comment by eruizdechavez** [comment]
@Arcanemagus I hear you!

**Comment by eruizdechavez** [comment]
@Arcanemagus and thanks for helping with this issue

**Comment by Arcanemagus** [comment]
I'm just glad it was traced down (probably, pending confirmation by @jeffschwartz)!

**Comment by jeffschwartz** [comment]
From the responses then it appears that a breaking change was introduced in v2.0.0. It cannot be jshint itself because it works with the previous version of the plugin.

**Comment by Arcanemagus** [comment]
@jeffschwartz Looking at the code there is a change related to this [here](https://github.com/AtomLinter/linter-jshint/compare/v1.3.1...v2.0.0#diff-2e1bb36917ff76ab7308da52bc968501L69), the `textEditor.getGrammar().scopeName.indexOf('text.html') isnt -1` check wasn't migrated over to [here](https://github.com/AtomLinter/linter-jshint/compare/v1.3.1...v2.0.0#diff-16b7d75b70953cbf4755508170f2f24cR103), which is causing this `jscs` bug to become exposed on a wider scale.

Do you want to submit a PR rectifying that?

**Comment by jeffschwartz** [comment]
I don't see how --extract=always has anything to do with this issue as the option only impacts in-line javascript and does not impact javascript in .js files. I haven't tried it but my suspicion tells me that linter-jshint is also broken for in-line javascript as well.

**Comment by jeffschwartz** [comment]
@Arcanemagus if I had the time, prior experience with CoffeeScript and a previous exposure to Atom development I would love to but I have none of those :(.

**Comment by Arcanemagus** [comment]
@jeffschwartz Check the last screenshot posted: Enabling that option completely breaks `jscs`.

The change between v1.3.1 and v2.0.0 is that previously when that option was enabled it would only call that when the file had the `text.html` scope, now it _always_ adds it.

I'll see if I have some time later to put up a PR for that.

**Comment by jeffschwartz** [comment]
@Arcanemagus I see that. Just curious but wouldn't a sanity/QA test have detected the issue and prevented the build from being released? Anyway, I really appreciate all the hard work you guys do in supporting and maintaining this plugin. It is immensely valuable in my development and I am sure I speak for many others who are using it as well :)

**Comment by Arcanemagus** [comment]
Of course it would, but this package has no active maintainer so it only has been getting brief bits of attention when @steelbrain or I have the time. A PR implementing specs would be wonderful :wink:

**Comment by Arcanemagus** [comment]
Can you check v2.0.1? It should be just as broken as v1.3.1 was :stuck_out_tongue:

The brokenness is within the `--extract` functionality though, so nothing we can really do about that here.

**Comment by jeffschwartz** [comment]
@Arcanemagus no tdd or bdd specs necessary for a sanity test/QA test. Just open a .js file in Atom with known errors and warnings. At that point, linting either works or it doesn't. Like I said previously, as I don't know CoffeeScript nor have I any Atom dev experience I wouldn't be the candidate to issue the pr for this since it should at least in my opinion be done ASAP because the production release of the plugin is broken. I really wish Atom had been developed using either pure JavaScript or TypeScript as that is where my expertise are. Again, my thanks go out to your and anyone else maintaining this project and I am sorry that I can't be of more help on the fix for this.

**Comment by eruizdechavez** [comment]
@Arcanemagus :+1:

**Comment by Arcanemagus** [comment]
@jeffschwartz There is no CoffeeScript in here as of v2.0.0, it's pure ES6 :wink:.

**Comment by jpoo90** [comment]
I'm having the same problem as @jeffschwartz the linter stopped working after I updated to v 2.0. Today I updated to v 2.0.2 but the problem is still there. 

I've tried restarting atom, enabling and disabling the linter without any luck. If I go back to v 1.3.1 it works again. 

I'm using Atom 1.5.3 with 
├── linter@1.11.3
├── linter-jshint@2.0.2 

Thanks in advance for any suggestions.

**Comment by eruizdechavez** [comment]
@jpoo90 have you tried to disable the option `Lint Inline Java Script` and see if it works?

**Comment by jpoo90** [comment]
@eruizdechavez I did try that, but it doesn't work either :/ 

Thanks for the suggestion.

**Comment by Arcanemagus** [comment]
@jpoo90 can you share your `jshintrc` file, an example file that you think should have errors but isn't showing anything, and the `linter-jshint` section of your configuration? (Or a screenshot of the Atom settings page for it works just as well for that)

**Comment by jpoo90** [comment]
**Config **

![settings](https://cloud.githubusercontent.com/assets/5958030/13123234/4eab4944-d589-11e5-8bf9-0e8fe840e990.png)

**.jshintrc **
`{
  "bitwise": true,
  "browser": true,
  "camelcase": true,
  "curly": true,
  "eqeqeq": true,
  "expr": false,
  "forin": true,
  "freeze": true,
  "immed": true,
  "indent": 2,
  "latedef": false,
  "newcap": true,
  "noarg": true,
  "node": true,
  "noempty": true,
  "nonew": false,
  "nonbsp": true,
  "plusplus": true,
  "quotmark": "single",
  "trailing": true,
  "undef": true,
  "unused": true,
  "strict": true,
  "trailing": true,
  "white": true,
  "jquery": true,
  "mocha": true,
  "globals": {
    "angular": true,
    "chai": true,  //testing lib
    "expect": true, //testing lib
    "sinon": true, //testing lib
    "_": false
  }
}`

**Test **
![test](https://cloud.githubusercontent.com/assets/5958030/13123705/7cf7e224-d58b-11e5-85bd-0970aa4c5224.gif)


@Arcanemagus Let me know if I can provide more feedback.

**Comment by Arcanemagus** [comment]
If you run `jshint search-module.js` from the command line do you get results? I'm not able to reproduce in a clean folder on it's own...

**Comment by jpoo90** [comment]
I get this results. 

```
src/modules/views/search/search-module.js: line 5, col 3, Expected '}' to match '{' from line 3 and instead saw 'b'.
src/modules/views/search/search-module.js: line 5, col 4, Missing semicolon.
src/modules/views/search/search-module.js: line 5, col 4, Unrecoverable syntax error. (62% scanned).
```

**Comment by Arcanemagus** [comment]
So something else is affecting your setup, as that's different from what I'm getting with your configuration file.
![image](https://cloud.githubusercontent.com/assets/427137/13124302/632390c0-d575-11e5-9fdf-dc3b72094387.png)

**Comment by jpoo90** [comment]
@Arcanemagus  We get the same errors. You have a different test, I just ran yours and got the same 11 errors. 

![screen shot 2016-02-17 at 4 39 53 pm](https://cloud.githubusercontent.com/assets/5958030/13125760/581a6e54-d595-11e5-92dd-9899de86b956.png)

**Comment by Arcanemagus** [comment]
Ah, I just pulled it from the GIF you posted. In any case obviously the configuration is not the issue here...

Can you open the Developer Tools within Atom and place breakpoints:

* [here](https://github.com/AtomLinter/linter-jshint/blob/v2.0.2/lib/main.js#L90) to ensure `lint()` is even being called.
* [here](https://github.com/AtomLinter/linter-jshint/blob/v2.0.2/lib/main.js#L114) and check the value of `output` to see the raw output `jshint` is giving us
* [here](https://github.com/AtomLinter/linter-jshint/blob/v2.0.2/lib/main.js#L125) and check the value of `parsed` to ensure the output has been parsed properly

**Comment by jpoo90** [comment]
![breakpoints](https://cloud.githubusercontent.com/assets/5958030/13126410/511844fc-d598-11e5-8b26-be4457008d1b.gif)

@Arcanemagus  It doesn't hit any breakpoints. Is there something I'm missing on the test ? Can I help with more feedback ?

**Comment by Arcanemagus** [comment]
If you run `Editor: Log Cursor Scope` from the command palette when your cursor is in the file, what scopes do you see listed?

**Comment by jpoo90** [comment]
![log cursor scope](https://cloud.githubusercontent.com/assets/5958030/13126798/01ba73d8-d59a-11e5-9c82-f33083ffb9aa.png)

**Comment by Arcanemagus** [comment]
Ah ha!

You have something installed (A language package perhaps?) that is marking _all_ `.js` files as JSX files. You either need to force the file to be recognized as JavaScript by clicking the language indicator in the lower right corner, or enable the "Lint JSX Files" option in the linter-jshint settings.

**Comment by jpoo90** [comment]
@Arcanemagus You're right I have the Babel Language Package. I activated the  `Lint JSX Files` and it is working now.

Thanks ! For your support, I appreciate it.

**Comment by Arcanemagus** [comment]
Glad we could get it figured out :wink:.
